frontstage_api/Resources/secure_messaging.py

```diff
@@ -17,10 +17,105 @@ class GetMessagesList(Resource):
+        if label == 'DRAFT':
+            draft = message
+            thread_id = draft.get('thread_id')
+            if thread_id != draft['msg_id']:
+                message = secure_messaging_controllers.get_thread_message(encoded_jwt, thread_id, party_id)
+            else:
+                message = {}
+        else:
+            draft = {}
+        # If the message was returned with errors return the error message
+        if message.get('error'):
+            make_response(jsonify(message), 200)
+
+        remove_unread_label = secure_messaging_controllers.remove_unread_label(encoded_jwt, message_id) if label == 'UNREAD' else {}
+        # Create json response
+        response_json = {
+            "message": message,
+            "draft": draft
+        }
+        # If we failed to remove the unread label append error but don't fail request
+        if remove_unread_label.get('error'):
+            response_json = {**response_json, **remove_unread_label}
+
+        return make_response(jsonify(response_json), 200)
+
+
+class SendMessage(Resource):
+    method_decorators = [get_jwt(request)]
+
+    @staticmethod
+    def post(encoded_jwt):
+        message_json = request.get_json(force=True)
+        party_id = message_json['msg_from']
+        is_draft = request.args.get('is_draft')
+
+        # Retrieving business party, case and survey id's
+        party = party_controller.get_party_by_respondent_id(party_id)
+        if party.get('error'):
+            return party
+        associations = party.get('associations')
+        if associations:
+            business_party_id = associations[0].get('partyId')
+            survey_id = associations[0].get('enrolments')[0].get('surveyId')
+
+        case = case_controller.get_case_by_party_id(party_id)
+        if not isinstance(case, list):
+            return case
+        case_id = case[0].get('id')
+
+        # Creating message json block to send to secure messaging
+        message_json = {
+            **message_json,
+            'msg_to': ['BRES'],
+            'msg_from': party_id,
+            'collection_case': case_id,
+            'ru_id': business_party_id,
+            'survey': survey_id
+        }
+
+        if is_draft == 'False':
+            message = secure_messaging_controllers.send_message(encoded_jwt, message_json)
+        else:
+            message = secure_messaging_controllers.save_draft(encoded_jwt, message_json)
+
+        # If the form was submitted with errors and is part of an existing thread, return the last message from thread
+        if message.get('error', {}).get('code') == 'FA006' and message_json.get('thread_id'):
+            thread_message = secure_messaging_controllers.get_thread_message(encoded_jwt, message_json['thread_id'], party_id)
+            message['error']['data'] = {**message['error']['data'], "thread_message": thread_message}
+
+        return make_response(jsonify(message), 200)
 
 
 api.add_resource(GetMessagesList, '/messages_list')
```

Comment:
It would be good if we could add the routes on the class. It makes it quite hard to follow if it's at the bottom

Reply:
done. I prefer that way too 👍